pip/req/req_install.py

```diff
@@ -237,7 +239,24 @@ def from_line(
                 req = Requirement(req)
             except InvalidRequirement:
                 if os.path.sep in req:
-                    add_msg = "It looks like a path. Does it exist ?"
+                    add_msg = "It looks like a path."
+                    if os.path.exists(req):
+                        add_msg += " It does exist."
+                        # Try to parse and check if it is a requirements file.
+                        try:
+                            with open(req, 'r') as fp:
+                                # parse first line only
+                                parse_requirements(fp.read()).next()
+                                add_msg += " The argument you provided " + \
+                                    "(%s) appears to be a" % (req) + \
+                                    " requirements file. If that is the" + \
+                                    " case, use the '-r' flag to install" + \
+                                    " the packages specified within it."
+                        except RequirementParseError:
+                            logger.debug("Cannot parse '%s' as requirements \
+                                file" % (req), exc_info=1)
+                    else:
+                        add_msg += " File '%s' does not exist." % (req)
```

Comment:
This starts to be a bit long and could be moved to its own function `deduce_helpful_msg()`